docs/source-pytorch/cli/lightning_cli_advanced.rst

```diff
@@ -60,6 +60,36 @@ with ``save_config_callback=None``.
 
         cli = LightningCLI(..., save_config_kwargs={"config_filename": "name.yaml"})
 
+It is also possible to extend the :class:`~lightning.pytorch.cli.SaveConfigCallback` class, for instance to additionally
+save the config in a logger. An example of this is:
+
+    .. code:: python
+
+        class LoggerSaveConfigCallback(SaveConfigCallback):
+            def save_config(self, trainer: Trainer, pl_module: LightningModule, stage: str) -> None:
+                if isinstance(trainer.logger, Logger):
+                    config = self.parser.dump(self.config, skip_none=False)  # Required for proper reproducibility
+                    trainer.logger.log_hyperparams({"config": config})
```

Comment:
@tommycwh I answer your [comment](https://github.com/Lightning-AI/lightning/issues/15340#issuecomment-1532397297) since it is about this pull request.

> However, I would want some more instructions on how to implement my save_config() function. For example, in my example above, there are some hyperparameters (like those numerical values) that can be saved by the "default" config saving mechanism and I would need to implement my own mechanism for the others that cannot be saved automaically (like the init_args).

I am not sure if I truly understand what you mean or what exactly is it that you want to achieve. This pull request adds `save_config` such that it can be used however people want to use it. In principle the config is intended for reproducibility and I would save it in a logger as an artifact or a single yaml string which includes any `init_args`. This is just three lines of code like shown in the example. Doing something different is perfectly fine, but it depends on what you want to do.